apps/spruce/src/components/Accordion/index.tsx

```diff
@@ -13,10 +13,13 @@ interface AccordionProps {
   showCaret?: boolean;
   title: React.ReactNode;
   titleTag?: React.FC;
+  /** `toggledTitle` replaces the title element when the accordion is open */
   toggledTitle?: React.ReactNode;
   toggleFromBottom?: boolean;
   useIndent?: boolean;
   subtitle?: React.ReactNode;
+  /** This prop prevents the accordion from rendering the child component if the accordion is collapsed. It is useful if the child is expensive to render. */
+  shouldRenderChildIfHidden?: boolean;
 }
 export const Accordion: React.FC<AccordionProps> = ({
```

Comment:
Could we align the caret at the top?
<img width="989" alt="Screenshot 2024-10-18 at 2 04 53 PM" src="https://github.com/user-attachments/assets/3365a879-08ee-4bef-afc0-0285361043e5">

The badge should probably be aligned too, or in-line with the text

Reply:
<img width="1138" alt="image" src="https://github.com/user-attachments/assets/c902fd13-d4ee-4bc0-87b2-8a3d19bd9ca0">
The badges are aligned with the test name. Aligning the caret to the top however seems more difficult. I can't come up with a way to modify the caret styling since that is controlled by the accordion without having it affect the styling for the rest of the accordions in which we want it to be center aligned. Open to suggestions!

Reply:
ooh.... I guess we could introduce a prop like `caretAlign`? kind of like how LeafyGreen has tooltip aligns and justify